src/Compilers/CSharp/Test/Emit2/Semantics/RefStructInterfacesTests.cs

```diff
@@ -0,0 +1,2914 @@
+
+class C : I
+{
+    [UnscopedRef]
+    public ref int M()
+    {
+        throw null;
+    }
+}
+";
+
+            foreach (var comp1Ref in comp1Refs)
+            {
+                var comp2 = CreateCompilation(src2, references: [comp1Ref], targetFramework: TargetFramework.Net80);
+                comp2.VerifyDiagnostics(
+                    // (6,6): error CS9101: UnscopedRefAttribute can only be applied to struct or virtual interface instance methods and properties, and cannot be applied to constructors or init-only members.
+                    //     [UnscopedRef]
+                    Diagnostic(ErrorCode.ERR_UnscopedRefAttributeUnsupportedMemberTarget, "UnscopedRef").WithLocation(6, 6)
+                    );
+                Assert.False(comp2.GetMember<MethodSymbol>("C.M").HasUnscopedRefAttribute);
+            }
+
+            var src3 = @"
+using System.Diagnostics.CodeAnalysis;
+
+class C : I
+{
+    [UnscopedRef]
+    ref int I.M()
+    {
+        throw null;
+    }
+}
+";
+
+            foreach (var comp1Ref in comp1Refs)
+            {
+                var comp3 = CreateCompilation(src3, references: [comp1Ref], targetFramework: TargetFramework.Net80);
+                comp3.VerifyDiagnostics(
+                    // (6,6): error CS9101: UnscopedRefAttribute can only be applied to struct or virtual interface instance methods and properties, and cannot be applied to constructors or init-only members.
+                    //     [UnscopedRef]
+                    Diagnostic(ErrorCode.ERR_UnscopedRefAttributeUnsupportedMemberTarget, "UnscopedRef").WithLocation(6, 6)
+                    );
+                Assert.False(comp3.GetMember<MethodSymbol>("C.I.M").HasUnscopedRefAttribute);
+            }
+
+            var src4 = @"
+class C1 : I
+{
+    int f = 0;
+    public ref int M()
+    {
+        return ref f;
+    }
+}
+
+class C2 : I
+{
+    int f = 0;
+    ref int I.M()
+    {
+        return ref f;
+    }
+}
+
+class C3
+{
+    int f = 0;
+    public ref int M()
+    {
+        return ref f;
+    }
+}
+
+class C4 : C3, I {}
+";
+
+            foreach (var comp1Ref in comp1Refs)
+            {
+                var comp4 = CreateCompilation(src4, references: [comp1Ref], targetFramework: TargetFramework.Net80);
+                CompileAndVerify(comp4, sourceSymbolValidator: verify, symbolValidator: verify, verify: Verification.Skipped).VerifyDiagnostics();
+
+                void verify(ModuleSymbol m)
+                {
+                    Assert.False(m.GlobalNamespace.GetMember<MethodSymbol>("C1.M").HasUnscopedRefAttribute);
+                    Assert.False(m.GlobalNamespace.GetMember<MethodSymbol>("C2.I.M").HasUnscopedRefAttribute);
+                    Assert.False(m.GlobalNamespace.GetMember<MethodSymbol>("C3.M").HasUnscopedRefAttribute);
+                }
+            }
+
+            var src5 = @"
+using System.Diagnostics.CodeAnalysis;
+
+interface C : I
+{
+    [UnscopedRef]
+    ref int I.M()
```

Comment:
Do we have a similar test (interface derived from another interface) but where the method is not explictly implemented?